### ISSUE TITLE
Allow the user to create samples for a new test order

### ISSUE THREAD

**Issue by nekron** [instedd/cdx#502]
The Id should be auto-numeric and should be re-set according to the interval specified in the Site settings.

Allow the user to specify a different id. Suggest the auto-numeric and allow the user to overwrite it.

See: https://projects.invisionapp.com/d/main#/console/4103589/93837626/preview https://projects.invisionapp.com/d/main#/console/4103589/113463974/preview and https://projects.invisionapp.com/d/main#/console/4103589/96562434/preview

**Comment by bcardiff** [comment]
Encounters/test orders do not belongs to a site right now.

* Should they start belonging to one? 
  * Mandatory?
* What should happen if the user is trying to create a test order and the navigation context is an institution but not a site?

The `sample_id_reset_policy` criteria depends on site and there is none associated to the institution per se.

* Probably related to a "scope test orders by selected site" story, but, should samples/test_results are ok not to scope their results by the selected site, but yes to the selected institution, right?
  * We should remove institution drop down from encounter page...

**Comment by nditada** [comment]
[Thinking out loud]: Test Orders are what a lab technician would use to know what things are in the pipeline and need to be processed. We then need a way to show them the ones that are meant for their site. Parts of the algorithm for a test order could require referring (sending) a sample to another lab, but that's out of scope for the MVP. 

In most cases test orders would be created by 'clerical' staff who would probably only have access to a single site. They shouldn't have to select any site. Users with access to more sites should have to select a site. 

So to summarize: 
>> Should they start belonging to one?
yes
>> Mandatory?
yes
>> What should happen if the user is trying to create a test order and the navigation context is an institution but not a site?
he sees dropdown for site selection. He should always see this (provided he has access for more than one site) even in cases where the current context is a site. The default should be the context site.

**Comment by bcardiff** [comment]
We need to handle the creation of the diagnose for test results that get added after the encounter was created. @spalladino / @nekron  I'm not sure if this should be done by the blender or when the encounter loaded from the UI.

**Comment by spalladino** [comment]
This is currently happening in the controller, at `#recalculate_diagnostic`, which is invoked whenever there is a change in the encounter's tests. Do we need something else to handle this?

**Comment by bcardiff** [comment]
@spalladino 
1. user creates test order
2. user add new fresh sample (without test results)
3. the sample is used in a device and a message arrives to the system
4. user loads the test order / encounter

expected: the diagnostic should appear. 
current behavior: empty diagnostic (user is unable to add assays manually).

options: 
a) update when loading the encounter so the user can validate the automatic change.
b) update them in the blender.

cons a) updated diagnostic is not available for search until the user enters and update each encounter. there is no flag for encounters to be reviwed.
cons b) we don't have yet a confirmed flag in the diagnosis. neither a diagnostic history in order to not lose the previous information.

**Comment by spalladino** [comment]
I'd go with option *b*. Actually, I believe it is a current bug as well, when you open an encounter autogenerated from an incoming test.

**Comment by mmuller** [comment]
Ok in version 0.11-pre6
